projects/atlendis-v2/index.js

```diff
@@ -1,57 +1,90 @@
-const { GraphQLClient, } = require("graphql-request");
+const { GraphQLClient } = require("graphql-request");
 const query = `
-  query tvl($chainId: Int!) {
+  query tvlAndBorrow($chainId: Int!) {
     rcls(chainId: $chainId) {
-      id
-     address
+      address
       tvl
       token {
         address
       }
-      currentLoan {accruedInterests borrowedAmount }
+      currentLoan {
+        borrowedAmount
+      }
     }
     bulletLoans(chainId: $chainId) {
-      id
-     address
+      address
       tvl
       token {
         address
       }
+      issuedLoan {
+        borrowedAmount
+      }
     }
   }
-`
+`;
 
 // Atlendis' indexer url
 const atlendisUrl = "https://atlendis.herokuapp.com/graphql";
 // Atlendis V2 currently supports two chains: Polygon and Mode
 const supportedChains = { polygon: 137, mode: 34443 };
 
 async function tvl(api) {
-  const chain = api.chain
+  const chain = api.chain;
   const graphQLClient = new GraphQLClient(atlendisUrl);
   const chainId = supportedChains[chain];
-  const { rcls, } = await graphQLClient.request(query, { chainId });
-  const tokensAndOwners = rcls.map(rcl => [rcl.token.address, rcl.address])
-  return api.sumTokens({ tokensAndOwners })
+  const { rcls, bulletLoans } = await graphQLClient.request(query, { chainId });
+  const balances = {};
+  for (let rcl of rcls) {
+    const poolTvl = rcl.tvl - rcl.currentLoan?.borrowedAmount || "0x00";
+    if (!balances[`${chain}:${rcl.token.address}`]) {
+      balances[`${chain}:${rcl.token.address}`] = poolTvl;
+    } else {
+      balances[`${chain}:${rcl.token.address}`] =
+        balances[`${chain}:${rcl.token.address}`] + poolTvl;
+    }
+  }
+  for (let loan of bulletLoans) {
+    const poolTvl = loan.tvl - loan.issuedLoan?.borrowedAmount || "0x00";
```

Comment:
still prefer counting only the token balance in contract as tvl. We try to maintain the same standard/logic for all protocols, sorry again about messing up the query.

BUt can you revert these parts to check token balance on contract like I had done